### PR TITLE
Skin Lesion detection using YOLOv3

### PR DESCRIPTION
**Pull request by lalithbv**
Hi Shakes (and COMP3710 team),

Please review my pull request containing the implementation of skin lesion detection within the ISICs dataset using YOLOv3 Object Detection with Darknet open-source neural network framework. The sample predictions of this model are provided in the Readme file in gif format and the detailed steps undertaken to build this model are also provided. The object detection performed with an average loss of 0.2545 and the mean average precision(mAP) of 95.78%.

The important files in my repository are:

- **Annotate_YOLO.py**: Creates "true" annotations labelled in the YOLO format. It takes in segmented(binary) images as input and returns annotations in the form of a text file for every image.
- **Helper.py**: Defines a helper function that is used to display images.
- **Generate_Train_Test.py**: This python script writes to "train.txt" and "test.txt", all the train and test image paths respectively.
- **obj.names**: Consists of the class name.
- **obj.data**: Points to the file paths of train data, validation data and backup folder which will store the weights of the model trained on our custom dataset.
- **yolov3-custom.cfg**: Custom config file created by me that tunes hyper-parameters of YOLO.

Thank you for providing me with the opportunity to work on an open-source project and to get hands-on experience on a state-of-the-art algorithm like YOLO. Please let me know if any aspect of this project could be improved or expanded upon, or if any issues need to be resolved.

Best Regards,

Lalith Veerabhadrappa Badiger
Student ID: 46557829

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
- Code appears okay for normal difficulty
- Commit messages are specific enough
- Performance is good

Other notes
- No driver code provided for using pre-trained Darknet repo

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Normal -5
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description good
